presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCountOverInnerAggregateQuery.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.facebook.presto.sql.tree.LongLiteral;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A count over a subquery can be reduced to a VALUES(1) provided
+ * the subquery is itself an aggregation
+ */
+public class SimplifyCountOverInnerAggregateQuery
+        implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator, Session session)
+    {
+        if (!(node instanceof AggregationNode)) {
+            return Optional.empty();
+        }
+        AggregationNode parent = (AggregationNode) node;
+        Map<Symbol, AggregationNode.Aggregation> assignments = parent.getAssignments();
+        if (assignments.size() > 1) {
+            return Optional.empty();
+        }
+        for (Map.Entry<Symbol, AggregationNode.Aggregation> entry : assignments.entrySet()) {
+            AggregationNode.Aggregation aggregation = entry.getValue();
+            if (aggregation != null) {
+                Signature signature = aggregation.getSignature();
+                if (!"count".equals(signature.getName())) {
```

Comment:
I think you need check also if `count` has arguments as `count(*)` has different semantic than `count(a)`